.github/workflows/container_build_template.yml

```diff
@@ -0,0 +1,193 @@
+        run: |
+          ANSIBLE_CORE_VERSION=$(cat ansible_collections/arista/avd/requirements-dev.txt| grep ansible-core)
+          echo "ansible_core_version=$ANSIBLE_CORE_VERSION" >> $GITHUB_OUTPUT
+
+      - name: find ansible install location
+        id: find-ansible-install-location
+        run: |
+          ANSIBLE_INSTALL_LOCATION="git+https://github.com/${{ github.repository }}.git#/ansible_collections/arista/avd/,${{ github.ref_name }}"
+          echo "ansible_install_location=$ANSIBLE_INSTALL_LOCATION" >> $GITHUB_OUTPUT
+
+      # This check is temporarily deactivated, will be rewised in later PRs
+      #
+      # - name: check if image is already present
+      #   # this step will check if image was already pushed before to avoid re-writing
+      #   # also if it's not a base image, this step will verify presence of the corresponding base image
+      #   id: check-image-presence
+      #   run: |
+      #     echo "not_present=0" >> $GITHUB_OUTPUT
+      #     for tag in $(echo ${{ steps.build-tags.outputs.image_tags }} | sed "s/,/ /g")
+      #     do
+      #       ( docker manifest inspect ghcr.io/${{ github.repository }}/${{ inputs.container_name }}:${tag} &>/dev/null ) || echo "not_present=1" >> $GITHUB_OUTPUT
+      #     done
+      #     # check if base image is present
+      #     if [ "${{ inputs.container_name }}" != "base" ]; then
+      #       echo "Verifying if base image is present..."
+      #       docker manifest inspect ghcr.io/${{ github.repository }}/base:python${{ inputs.python_version }} >/dev/null
+      #     fi
+
+      - name: setup QEMU for multi-arch builds 🏗️
+        # if: ${{ steps.check-image-presence.outputs.not_present == '1' }}
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: ${{ inputs.platform }}
+
+      - name: setup Docker buildX for multi-arch builds 🏗️
+        # if: ${{ steps.check-image-presence.outputs.not_present == '1' }}
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to Docker Hub 🗝️
+        # if: ${{ steps.check-image-presence.outputs.not_present == '1' }}
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Pre-build dev container image 🔨
+        # if: ${{ steps.check-image-presence.outputs.not_present == '1' }}
+        uses: devcontainers/ci@v0.3
+        env:
+          FROM_IMAGE: ${{ steps.generate-from-parameters.outputs.from_image }}
+          FROM_VARIANT: ${{ steps.generate-from-parameters.outputs.from_variant }}
+          ANSIBLE_INSTALL_LOCATION: ${{ steps.find-ansible-install-location.outputs.ansible_install_location }}
+          ANSIBLE_CORE_VERSION: ${{ steps.find-ansible-core-version.outputs.ansible_core_version }}
+          USERNAME: ${{ inputs.username }}
+          UID: ${{ inputs.user_id }}
+          GID: ${{ inputs.group_id }}
```

Comment:
is the strategy to make this configurable only at build time for now? (it is fine for me though we may need to build a special one with uid 1001 for github runner)

Reply:
Normally I don't expect this to be changed. But I'd like to avoid configuring any hardcoded parameters in container definitions. All parameters must come from the workflow even if changing them is not realistic.